File: data-automation-project/backend/data-automation-project/src/main/java/com/studies/lvb/data_automation_project/service/UserService.java
package com.studies.lvb.data_automation_project.service;

import com.studies.lvb.data_automation_project.dto.UserDTO;
import com.studies.lvb.data_automation_project.model.User;
import com.studies.lvb.data_automation_project.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PasswordEncoder passwordEncoder;

    public UserDTO createUser(final UserDTO userDto) {

        //Encrypt password
        final String encryptedPassword = passwordEncoder.encode(userDto.getPassword());

        final User user = User.builder()
                .name(userDto.getName())
                .email(userDto.getEmail())
                .encryptedPassword(encryptedPassword)
                .build();

        // Save user
        userRepository.save(user);

        return userDto;
    }
}

File: BE-Dev-Courses/Java/Spring JPA, RESTFul, Security, JWT/point-of-sale/point-of-sale-api/basic-auth/src/main/kotlin/com/lvb/studys/kotlin/api/pointofsaleapi/basicauth/service/UserService.kt
package com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.service

import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.dto.UserDTO
import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.dto.UserResponseDTO
import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.entity.User
import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.exceptions.NoItemException
import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.repository.UserRepository
import com.lvb.studys.kotlin.api.pointofsaleapi.basicauth.security.SecurityConfig
import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class UserService @Autowired constructor(
    private val userRepository: UserRepository
) {

    private var modelMapper: ModelMapper = ModelMapper()

    fun getAllUsers(): List<UserResponseDTO> {
        return userRepository.findAll().map {
            UserResponseDTO(it.id, it.name, it.isEnabled, it.username)
        }.toList()
    }

    fun findUserById(id: Long): UserDTO {
        val user = userRepository.findById(id)

        if (user.isEmpty) {
            throw NoItemException("User not found")
        }

        return UserDTO(user.get().id, user.get().name, user.get().isEnabled, user.get().username, user.get().password)

    }


    fun saveUser(user: UserDTO): UserDTO {
        val newUser = modelMapper.map(user, User::class.java)

        newUser.isEnabled = true
        newUser.password = SecurityConfig.passwordEncoder().encode(user.password)

        userRepository.save(newUser)
        return UserDTO(newUser.id, newUser.name, newUser.isEnabled, newUser.username, newUser.password)
    }

    fun updateUser(user: UserDTO): UserDTO {
        findUserById(user.id!!)

        val userToUpdate = modelMapper.map(user, User::class.java)
        userToUpdate.password = SecurityConfig.passwordEncoder().encode(user.password)

        userRepository.save(userToUpdate)
        return user
    }

    fun deleteUser(id: Long) {
        userRepository.deleteById(id)
    }

    fun getByUsername(username: String) : User = userRepository.findUserByUsername(username)
}